src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1343 @@ ActorInit Boss_05_InitVars = {
+    } else if (BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_WALKING_HEAD) {
+        Boss05_WalkingHead_SetupTransform(this, play);
+        this->dyna.actor.colChkInfo.mass = 90;
+
+        SkelAnime_InitFlex(play, &this->headSkelAnime, &gBioDekuBabaHeadSkel, &gBioDekuBabaHeadChompAnim,
+                           this->headJointTable, this->headMorphTable, BIO_DEKU_BABA_HEAD_LIMB_MAX);
+
+        Collider_InitAndSetJntSph(play, &this->headCollider, &this->dyna.actor, &sWalkingHeadJntSphInit,
+                                  this->headColliderElements);
+
+        ActorShape_Init(&this->dyna.actor.shape, 0.0f, ActorShadow_DrawCircle, 30.0f);
+        this->dyna.actor.colChkInfo.damageTable = &sWalkingHeadDamageTable;
+        this->dyna.actor.flags |= ACTOR_FLAG_10 | ACTOR_FLAG_20;
+    } else if (BIO_BABA_GET_TYPE(&this->dyna.actor) >= BIO_BABA_TYPE_FRAGMENT_BASE) {
+        SkelAnime_InitFlex(play, &this->headSkelAnime, &gBioDekuBabaHeadSkel, &gBioDekuBabaHeadChompAnim,
+                           this->headJointTable, this->headMorphTable, BIO_DEKU_BABA_HEAD_LIMB_MAX);
+
+        this->dyna.actor.gravity = 0.0f;
+        this->dyna.actor.world.rot.y = Rand_ZeroFloat(0x8000);
+        this->dyna.actor.speed = Rand_ZeroFloat(3.0f) + 3.0f;
+        this->dyna.actor.velocity.y = Rand_ZeroFloat(1.5f) + 1.5f;
+
+        this->fragmentAngularVelocity.x = Rand_CenteredFloat(700.0f);
+        this->fragmentAngularVelocity.y = Rand_CenteredFloat(1500.0f);
+        this->timers[TIMER_CURRENT_ACTION] = Rand_ZeroFloat(30.0f) + 50.0f;
+
+        this->dyna.actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        this->actionFunc = Boss05_Fragment_Move;
+    }
+}
+
+void Boss05_Destroy(Actor* thisx, PlayState* play) {
+    Boss05* this = THIS;
+
+    if ((BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_LILY_PAD) ||
+        (BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_LILY_PAD_WITH_HEAD) ||
+        (BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_NO_LEAF_LILY_PAD_WITH_HEAD)) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+/**
+ * Checks to see if the lily pad stem or the head was hit by an attack. If either of them were hit, this function will
+ * return a value indicating what the head should do; if the head should detach from the lily pad, then the draw damage
+ * effect to apply to the falling head will be included in the return value.
+ */
+s32 Boss05_LilyPadWithHead_UpdateDamage(Boss05* this, PlayState* play) {
+    if (this->damagedFlashTimer == 0) {
+        s32 i = 0;
+
+        while (true) {
+            if (this->lilyPadCollider.elements[i].info.bumperFlags & BUMP_HIT) {
+                switch (this->dyna.actor.colChkInfo.damageEffect) {
+                    case BIO_BABA_DMGEFF_FIRE:
+                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_FIRE_INIT;
+                    case BIO_BABA_DMGEFF_FREEZE:
```

Comment:
```suggestion
                    case BIO_BABA_DMGEFF_FIRE:
                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_FIRE_INIT;
                        
                    case BIO_BABA_DMGEFF_FREEZE:
```
Could you space out this switch case?